### PR TITLE
Add "system_settings" config doc

### PR DESCRIPTION
**Pull request by johnliu55tw**
**Note:** We could add a reference link to "Settings" page added by PR https://github.com/harvester/docs/pull/73 once it's merged.

image

### CODE REVIEW

File: docs/install/harvester-configuration.md [hunk 1]
```diff
@@ -425,3 +427,24 @@ install:
...
+#### Definition
+
+Use this config to overwrite default Harvester system settings.
+
+!!! note
+    Overwriting system settings only works when Harvester is installing in "create" mode. When installing in "join" mode, this config has no effect.
+
+#### Example
+
+Automatically adding storage devices on every Nodes that match the glob pattern as VM storages.
```
Comment (nunix) [inline docs/install/harvester-configuration.md:442]:
```suggestion
The configuration below will automatically scan and add storage devices which match the glob pattern `/dev/sd*` on every Node.
```

**Review by nunix — COMMENTED** [review]
Rephrased the definition, note and example description

File: docs/install/harvester-configuration.md [hunk 2]
```diff
@@ -425,3 +427,24 @@ install:
   vip_mode: dhcp
   vip_hw_addr: 52:54:00:ec:0e:0b
 ```
+
+### `system_settings`
+
+#### Definition
+
+Use this config to overwrite default Harvester system settings.
```
[the remark this answers is not in the document]
Comment (johnliu55tw) [reply docs/install/harvester-configuration.md]: Rephrased

File: docs/install/harvester-configuration.md [hunk 3]
```diff
@@ -425,3 +427,24 @@ install:
...
+
+!!! note
+    Overwriting system settings only works when Harvester is installing in "create" mode. When installing in "join" mode, this config has no effect.
```
[the remark this answers is not in the document]
Comment (johnliu55tw) [reply docs/install/harvester-configuration.md]: Rephrased

File: docs/install/harvester-configuration.md (hunk 1 above)
Comment (johnliu55tw) [reply docs/install/harvester-configuration.md]: Rephrased

**Review by nunix — COMMENTED** [review]
LGTM

File: docs/install/harvester-configuration.md [hunk 4]
```diff
@@ -425,3 +427,27 @@ install:
   vip_mode: dhcp
   vip_hw_addr: 52:54:00:ec:0e:0b
 ```
+
+### `system_settings`
+
+#### Definition
+
+This setting allows you to overwrite the default Harvester system settings.
```
Comment (nunix) [inline docs/install/harvester-configuration.md:435]:
```suggestion
You can overwrite the default Harvester system settings by configuring `system_settings`.
```

File: docs/install/harvester-configuration.md [hunk 5]
```diff
@@ -425,3 +427,27 @@ install:
...
+
+!!! note
+    Overwriting system settings is only applicable if Harvester is installed in "create" mode.
+    Instead, if the "join" mode is selected during the Harvester installation, this setting will
+    be discarded.
```
Comment (nunix) [inline docs/install/harvester-configuration.md:440]:
```suggestion
    Overwriting system settings only works when Harvester is installed in "create" mode.
    If you install Harvester in "join" mode, this setting is ignored.
```

File: docs/install/harvester-configuration.md [hunk 6]
```diff
@@ -425,3 +427,27 @@ install:
...
+
+#### Example
+
+The configuration below will automatically scan and add storage devices which match the glob
+pattern `/dev/sd*` on every Node.
+
+!!! warning
+    All the data in these storage devices will be destroyed. Use it at your own risk.
```
Comment (nunix) [inline docs/install/harvester-configuration.md:448]:
```suggestion
    All the data in these storage devices will be destroyed. Use at your own risk.
```

File: docs/install/harvester-configuration.md [hunk 7]
```diff
@@ -425,3 +427,27 @@ install:
...
+
+!!! note
+    Overwriting system settings is only applicable if Harvester is installed in "create" mode.
+    Instead, if the "join" mode is selected during the Harvester installation, this setting will
+    be discarded.
+
+#### Example
+
+The configuration below will automatically scan and add storage devices which match the glob
+pattern `/dev/sd*` on every Node.
```
Comment (nunix) [inline docs/install/harvester-configuration.md:445]:
```suggestion
The configuration below will automatically scan and add storage devices which will match the glob
pattern `/dev/sd*` on every Node.
```

**Review by nunix — COMMENTED** [review]
Rephrased based on Jen feedback on Slack

File: docs/install/harvester-configuration.md (hunk 5 above)
Comment (johnliu55tw) [reply docs/install/harvester-configuration.md:440]: In addition to the original words, I'd like to emphasize one thing (The bolded sentence):
> ..., this setting is ignored. **All the settings on the server ("create") Node will be applied to agent ("join") Nodes automatically.**

@nunix please also help check on this.

**Review by bk201 — APPROVED** [review]

File: docs/install/harvester-configuration.md (hunk 5 above)
Comment (guangbochen) [reply docs/install/harvester-configuration.md:440]: @johnliu55tw we don't emphasize the difference between server or agent nodes a lot for normal users, so prefer to use the `create`  or `join` mode only.

Comment (johnliu55tw) [reply docs/install/harvester-configuration.md:440]: Good idea, I'll rephrase it a bit.

**Review by nunix — APPROVED** [review]

**Comment by guangbochen** [comment]
@johnliu55tw can we squash the commits, thanks.

**Comment by johnliu55tw** [comment]
@guangbochen done. Also since PR https://github.com/harvester/docs/pull/73 is merged, I will add "auto disk provision" and "SSL Parameters" settings later.

**Pull request merged**